### PR TITLE
feat(frontend): Add sql string into context for debugging

### LINKED ISSUE #2777: frontend: Store sql in current execution context for better debugging.
**Is your feature request related to a problem? Please describe.**

Currently after parsing, our raw sql is discarded. This make printing debugging information including sql impossible. 

**Describe the solution you'd like**

Store raw sql, in current execution context or session to so that following debugging will be easier.

**Describe alternatives you've considered**

No.

**Additional context**
No.

### PR DESCRIPTION
**Pull request by nhzaci**
## What's changed and what's your intention?

Add `Arc<str>` into `OptimizerContext` to store `raw_sql` strings for debugging.

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)

  - Add `Arc<str>` into `OptimizerContext`
  - Refactor to create `Arc<str>` at entry points

- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)
- Add the 'user-facing changes' label if your PR contains changes that are visible to users (optional)

## Checklist

- [X] I have written necessary docs and comments
- [X] I have added necessary unit tests and integration tests
- [X] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

Closes #2777

### CODE REVIEW

File: src/frontend/src/session.rs [hunk 1]
```diff
@@ -62,6 +62,8 @@ pub struct OptimizerContext {
     pub session_ctx: Arc<SessionImpl>,
     // We use `AtomicI32` here because  `Arc<T>` implements `Send` only when `T: Send + Sync`.
     pub next_id: AtomicI32,
+    /// For debugging purposes, store the SQL string in Context
+    pub sql: Arc<String>,
```
Comment (BugenZhao) [inline src/frontend/src/session.rs:66]: We should prefer `Arc<str>` over `Arc<String>`.

**Review by BugenZhao — COMMENTED** [review]
Thanks for your contribution!

File: src/frontend/src/session.rs (hunk 1 above)
Comment (nhzaci) [reply src/frontend/src/session.rs]: Thanks, updated to use `str` instead of `String`

**Review by lmatz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**